swupd/packs.go

```diff
@@ -232,10 +242,27 @@ func WritePack(w io.Writer, fromManifest, toManifest *Manifest, outputDir, chroo
 			return nil, err
 		}
 
+		dManifest.Files = append(dManifest.Files, d.from)
 		info.DeltaCount++
 		hasDelta[d.to.Hash] = d
 	}
 
+	if fromManifest != nil {
+		dManifest.Header.FileCount = uint32(len(dManifest.Files))
+		if dManifest.Header.FileCount > 0 {
+			dManifestPath := filepath.Join(outputDir, fmt.Sprint(toVersion), "delta",
+				fmt.Sprintf("Manifest.%s.D.%d", dManifest.Name, fromVersion))
+
+			if err = dManifest.WriteManifestFile(dManifestPath); err != nil {
+				return nil, err
+			}
+
+			if _, err = copyFromPath(tw, dManifestPath, "delta/"); err != nil {
```

Comment:
Not entirely sure I love throwing this in `delta/` and I'm not sure `swupd` will handle this gracefully enough. Since `swupd` iterates through everything in `<statedir>/delta` I believe it will print warnings that it `couldn't use delta <delta manifest>` for every delta manifest it gets. Can we put the delta manifest in the top level of the delta pack instead?
```
delta/
    <delta files>
staged/
    <full files>
Manifest.os-core.D.24710
```

Reply:
Good point. But if we do that, we are going to have Delta manifests listed in /var/lib/swupd. Is this a problem?

And I realized by your comment that we have another issue with this PR. We may have, for example, Manifest.os-core.D.10 in version 20 and 30. If we download them both (I don't see a reason to do that, but this may happen in some combinations of swupd updates), one file will replace the other because they are extracted in the same directory. What about renaming them to Manifest.<bundle>.D.<from_version>.<to_version>.

I'll check if swupd isn't breaking with this PR and if we need a format bump. I've done that with IManifests, but not for delta manifests.

Reply:
So currently individual manifests are placed under `/var/lib/swupd/<version>/Manifest.<bundle>`, so you could potentially do the same thing here. The packs would then be constructed by creating a new directory `<version>` containing the delta manifest. This way it will be sent to the correct location on the client when extracted.

I'd prefer that approach to placing them at the top-level because it keeps the structure of the cache nicer and places the delta manifests alongside the full manifests.

On the client side these delta manifests should probably be cleaned up on updates since they are no longer valid after being used (if the client updated).

Reply:
I was wondering a good way to do that on swupd, but it makes a lot of sense to create a <version> folder on the pack. Good idea.

So packs content (for your example) should be like this:

```
delta/
    <delta files>
staged/
    <full files>
25720/
     Manifest.os-core.D.24710
```

Reply:
Correct